persistence-modules/spring-data-jpa-2/src/test/java/com/baeldung/aggregation/SpringDataAggregateIntegrationTest.java

```diff
@@ -0,0 +1,81 @@
+package com.baeldung.aggregation;
+
+import com.baeldung.aggregation.model.custom.CommentCount;
+import com.baeldung.aggregation.model.custom.ICommentCount;
+import com.baeldung.aggregation.repository.CommentRepository;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(SpringRunner.class)
+@DataJpaTest
+@Sql(scripts = "/test-aggregation-data.sql")
+public class SpringDataAggregateIntegrationTest {
+
+    @Autowired
+    private CommentRepository commentRepository;
+
+    @Test
+    public void whenQueryWithAggregation_thenReturnResult() {
+        List<Object[]> commentCountsByYear = commentRepository.countTotalCommentsByYear();
+        Object[] countYear2019 = commentCountsByYear.get(0);
+        assertEquals(countYear2019[0], Integer.valueOf(2019));
+        assertEquals(countYear2019[1], 1l);
+        Object[] countYear2018 = commentCountsByYear.get(1);
+        assertEquals(countYear2018[0], Integer.valueOf(2018));
+        assertEquals(countYear2018[1], 2l);
+        Object[] countYear2017 = commentCountsByYear.get(2);
+        assertEquals(countYear2017[0], Integer.valueOf(2017));
+        assertEquals(countYear2017[1], 1l);
+    }
+
+    @Test
+    public void whenQueryWithAggregation_thenReturnCustomResult() {
+        List<CommentCount> commentCountsByYear = commentRepository.countTotalCommentsByYearClass();
+        CommentCount countYear2019 = commentCountsByYear.get(0);
+        assertEquals(countYear2019.getYear(), Integer.valueOf(2019));
```

Comment:
This assertion is backwards. It should be `assertEquals(expected, actual)`

Reply:
I changed the assertions to use assertThat.